From 4d17f2c3b5d19a47548945527f0058128b421934 Mon Sep 17 00:00:00 2001
From: Jun Miao <jun.miao@windriver.com>
Date: Tue, 25 Feb 2020 14:01:24 +0800
Subject: [PATCH] Revert "Revert "net: Use TC_SETUP_QDISC_MQPRIO""

This reverts commit ab28e2d0224534b67246a34b7001c25f6454937b.
Besides, dpaa2-eth.c rename the TC_SETUP_MQPRIO to TC_SETUP_QDISC_MQPRIO.
dpaa-eth.c driver need to use the TC_SETUP_QDISC_MQPRIO.

Signed-off-by: Jun Miao <jun.miao@windriver.com>
---
 drivers/net/ethernet/amd/xgbe/xgbe-drv.c       | 2 +-
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c | 2 +-
 include/linux/netdevice.h                      | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/amd/xgbe/xgbe-drv.c b/drivers/net/ethernet/amd/xgbe/xgbe-drv.c
index 093e2fd1208b..24f1053b8785 100644
--- a/drivers/net/ethernet/amd/xgbe/xgbe-drv.c
+++ b/drivers/net/ethernet/amd/xgbe/xgbe-drv.c
@@ -2253,7 +2253,7 @@ static int xgbe_setup_tc(struct net_device *netdev, enum tc_setup_type type,
 	struct tc_mqprio_qopt *mqprio = type_data;
 	u8 tc;
 
-	if (type != TC_SETUP_MQPRIO)
+	if (type != TC_SETUP_QDISC_MQPRIO)
 		return -EOPNOTSUPP;
 
 	mqprio->hw = TC_MQPRIO_HW_OFFLOAD_TCS;
diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index 385fc6968627..7d852b8ce032 100755
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -1869,7 +1869,7 @@ static int dpaa2_eth_setup_tc(struct net_device *net_dev,
 	struct tc_mqprio_qopt *mqprio = (struct tc_mqprio_qopt *)type_data;
 	int i, err = 0;
 
-	if (type != TC_SETUP_MQPRIO)
+	if (type !=TC_SETUP_QDISC_MQPRIO)
 		return -EINVAL;
 
 	if (mqprio->num_tc > dpaa2_eth_tc_count(priv)) {
diff --git a/include/linux/netdevice.h b/include/linux/netdevice.h
index 80c09e22779e..2e5800802bdb 100644
--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -798,7 +798,7 @@ typedef u16 (*select_queue_fallback_t)(struct net_device *dev,
 				       struct sk_buff *skb);
 
 enum tc_setup_type {
-	TC_SETUP_MQPRIO,
+	TC_SETUP_QDISC_MQPRIO,
 	TC_SETUP_CLSU32,
 	TC_SETUP_CLSFLOWER,
 	TC_SETUP_CLSMATCHALL,
-- 
2.17.1

